ros2param/ros2param/verb/list.py

```diff
@@ -69,53 +72,20 @@ def main(self, *, args):  # noqa: D102
         if regex_filter is not None:
             regex_filter = re.compile(regex_filter[0])
 
-        with NodeStrategy(args) as node:
-            service_names = get_service_names(
-                node=node, include_hidden_services=args.include_hidden_nodes)
-
         with DirectNode(args) as node:
-            clients = {}
-            futures = {}
-            # create clients for nodes which have the service
+            responses = {}
             for node_name in node_names:
-                service_name = f'{node_name.full_name}/list_parameters'
-                if service_name in service_names:
-                    client = node.create_client(ListParameters, service_name)
-                    clients[node_name] = client
-
-            # wait until all clients have been called
-            while True:
-                for node_name in [
-                    n for n in clients.keys() if n not in futures
-                ]:
-                    # call as soon as ready
-                    client = clients[node_name]
-                    if client.service_is_ready():
-                        request = ListParameters.Request()
-                        for prefix in args.param_prefixes:
-                            request.prefixes.append(prefix)
-                        future = client.call_async(request)
-                        futures[node_name] = future
-
-                if len(futures) == len(clients):
-                    break
-                rclpy.spin_once(node, timeout_sec=1.0)
-
-            # wait for all responses
-            for future in futures.values():
-                rclpy.spin_until_future_complete(node, future, timeout_sec=1.0)
-
+                responses[node_name] = call_list_parameters(node=node, node_name=node_name.name)
```

Comment:
I think we should use the fully-qualified node name here.
```suggestion
                responses[node_name] = call_list_parameters(node=node, node_name=node_name.full_name)
```